server/src/main/java/org/opensearch/search/query/QueryPhase.java

```diff
@@ -308,7 +309,32 @@ static boolean executeInternal(SearchContext searchContext) throws QueryPhaseExe
                 if (sortAndFormatsForRewrittenNumericSort != null && collectors.size() == 0 && searchContext.getProfilers() == null) {
                     shouldRescore = searchWithCollectorManager(searchContext, searcher, query, leafSorter, timeoutSet);
                 } else {
-                    shouldRescore = searchWithCollector(searchContext, searcher, query, collectors, hasFilterCollector, timeoutSet);
+                    boolean couldUseConcurrentSegmentSearch = searcher.allowConcurrentSegmentSearch();
+
+                    // TODO: support aggregations
+                    if (searchContext.aggregations() != null) {
+                        couldUseConcurrentSegmentSearch = false;
```

Comment:
ah I see, the concurrency is disabled for aggregations, so that previous exception should never be thrown